applications/shape_detector/src/plane_detector.py

```diff
@@ -0,0 +1,939 @@
+
+#     if  not singular :
+#         x = np.arctan2(R[2,1] , R[2,2])
+#         y = np.arctan2(-R[2,0], sy)
+#         z = np.arctan2(R[1,0], R[0,0])
+#     else :
+#         x = np.arctan2(-R[1,2], R[1,1])
+#         y = np.arctan2(-R[2,0], sy)
+#         z = 0
+
+#     theta = np.rad2deg(np.array([x, y, z]))
+#     return theta
+
+
+# # Calculates Rotation Matrix given euler angles.
+# def eulerAnglesToRotationMatrix(theta) :
+
+#     theta = np.deg2rad(theta)
+ 
+#     R_x = np.array([[1,         0,                  0                   ],
+#                     [0,         np.cos(theta[0]), -np.sin(theta[0]) ],
+#                     [0,         np.sin(theta[0]), np.cos(theta[0])  ]
+#                     ])
+ 
+#     R_y = np.array([[np.cos(theta[1]),    0,      np.sin(theta[1])  ],
+#                     [0,                     1,      0                   ],
+#                     [-np.sin(theta[1]),   0,      np.cos(theta[1])  ]
+#                     ])
+ 
+#     R_z = np.array([[np.cos(theta[2]),    -np.sin(theta[2]),    0],
+#                     [np.sin(theta[2]),    np.cos(theta[2]),     0],
+#                     [0,                     0,                      1]
+#                     ])
+ 
+#     R = np.dot(R_z, np.dot( R_y, R_x ))
+ 
+#     return R
+
+#%% Main
+class PlaneDetector:
+    def __init__(self):
+
+        self.frame_size = (640,480)
+        self.img        = None
+        self.cam_matrix = np.array([[1000,0,self.frame_size[0]/2],[0,1000,self.frame_size[1]/2],[0,0,1]], dtype = np.float32)
+        self.cam_distort= np.array([0,0,0,0,0],dtype = np.float32)
+
+        self.img3d      = None # contains x,y and depth plains
+        self.imgXYZ     = None  # comntains X,Y,Z information after depth image to XYZ transform
+        self.imgMask    = None  # which pixels belongs to which cluster
+
+        # params
+        self.MIN_SPLIT_SIZE  = 32
+        self.MIN_STD_ERROR   = 0.01
+
+        # help variable
+        self.ang_vec     = np.zeros((3,1))  # help variable
+
+    def add_noise(self, img_gray, noise_percentage = 0.01):
+        "salt and pepper noise"
+        if noise_percentage < 0.001:
+            return img_gray
+
+
+        # Get the image size (number of pixels in the image).
+        img_size = img_gray.size
+
+        # Set the percentage of pixels that should contain noise
+        #noise_percentage = 0.1  # Setting to 10%
+
+        # Determine the size of the noise based on the noise precentage
+        noise_size = int(noise_percentage*img_size)
+
+        # Randomly select indices for adding noise.
+        random_indices = np.random.choice(img_size, noise_size)
+
+        # Create a copy of the original image that serves as a template for the noised image.
+        img_noised = img_gray.copy()
+
+        # Create a noise list with random placements of min and max values of the image pixels.
+        #noise = np.random.choice([img_gray.min(), img_gray.max()], noise_size)
```

Comment:
please check if the right line is uncommented and eventually add a comment to explain the -10, 10